api/app/responses/tests.py

```diff
@@ -85,270 +94,268 @@ def _seed_data(self):
 
     def test_get_response(self):
         """Test a typical GET flow."""
-        with app.app_context():
-            self._seed_data()
-
-            response = self.app.get('/api/v1/response',
-                                    headers={
-                                        'Authorization': self.other_user_data['token']},
-                                    query_string={'event_id': self.test_event.id})
-            data = json.loads(response.data)
-
-            self.assertEqual(data['application_form_id'], self.test_form.id)
-            self.assertEqual(data['user_id'], self.other_user_data['id'])
-            self.assertIsNone(data['submitted_timestamp'])
-            self.assertFalse(data['is_withdrawn'])
-            self.assertTrue(data['answers'])
-
-            answer = data['answers'][0]
-            self.assertEqual(answer['id'], self.test_answer1.id)
-            self.assertEqual(answer['value'], self.test_answer1.value)
-            self.assertEqual(answer['question_id'], self.test_question.id)
+        
+        self._seed_data()
+
+        response = self.app.get('/api/v1/response',
+                                headers={
+                                    'Authorization': self.other_user_data['token']},
+                                query_string={'event_id': self.test_event.id})
+        data = json.loads(response.data)
+
+        self.assertEqual(data['application_form_id'], self.test_form.id)
+        self.assertEqual(data['user_id'], self.other_user_data['id'])
+        self.assertIsNone(data['submitted_timestamp'])
+        self.assertFalse(data['is_withdrawn'])
+        self.assertTrue(data['answers'])
+
+        answer = data['answers'][0]
+        self.assertEqual(answer['id'], self.test_answer1.id)
+        self.assertEqual(answer['value'], self.test_answer1.value)
+        self.assertEqual(answer['question_id'], 1)
```

Comment:
So I kept getting an error along the lines of
`DetachedInstanceError: Instance <Event at 0x7fdb85c18710> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/bhk3)` but only if I ran the tests all at once. Running a test one at a time, they didn't break.

This is why I changed this whole file. Seems it's something to do with the objects not being loaded correctly anymore. Removing the app.context() and hard coding some of the values fixed the issue. Probably not the best way of fixing it but was a bit stumped so if someone knows the proper fix, let me know.

Reply:
I had the same issue the other day, and also couldn't figure out what was causing it :( @jaderabbit any idea?